mri/operators/fourier/non_cartesian.py

```diff
@@ -494,18 +498,25 @@ def __init__(self, kspace_loc, shape, implementation='cpu', n_coils=1):
     def _op(self, data):
         fft_along_z_axis = np.fft.fftshift(np.fft.fft(
             np.fft.ifftshift(data, axes=2),
-            n=self.acq_num_slices,
             norm="ortho"),
             axes=2)
+
         stacked_kspace = np.asarray(
-            [self.plane_fourier_operator.op(fft_along_z_axis[:, :, slice])
-             for slice in np.arange(self.acq_num_slices)])
-        stacked_kspace = np.reshape(stacked_kspace,
-                                    self.acq_num_slices * self.stack_len)
+            [self.plane_fourier_operator.op(fft_along_z_axis[:, :, stack])
+             for stack in range(self.num_slices)])
+        stacked_kspace_sampled = np.zeros((self.acq_num_slices,
+                                           self.stack_len),
+                                          dtype=data.dtype)
+        stacked_kspace_sampled = stacked_kspace[self.idx_mask_z]
```

Comment:
Please correct me if my understanding here is wrong:
You are effectively doing a complete 2D NFFT on each slice location and choosing undersampled data.
Can we move to just do NFFT on only the slices where we want?

Something like:

```python
stacked_kspace_sampled = np.asarray(
            [self.plane_fourier_operator.op(fft_along_z_axis[:, :, stack])
             for stack in self.idx_mask_z])
```

Was this explored and would this cause any issues?

Reply:
I made the change, thank you